src/Subsampling/include/gudhi/choose_n_farthest_points.h

```diff
@@ -142,6 +143,217 @@ void choose_n_farthest_points(Distance dist,
+  std::vector<std::size_t> modified_neighbors;
+  boost::unordered_set<std::size_t> l_neighbors; // Should we use an allocator?
+  for (std::size_t current_number_of_landmarks = 1; current_number_of_landmarks != final_size; current_number_of_landmarks++) {
+    std::size_t l_parent = radius_priority.top();
+    auto& parent_info = landmarks[l_parent];
+    std::size_t l = parent_info.far;
+    FT radius = parent_info.radius;
+    auto& info = landmarks[l];
+    *output_it++ = input_pts[l];
+    *dist_it++ = radius;
+    l_neighbors.clear();
+    modified_neighbors.clear();
+    // If a Voronoi point X of A can steal a Voronoi point Y from B, then
+    // BY > XY >= AB - AX - BY, so AB < AX + 2 * BY. Symmetrized.
+    auto max_dist = [](FT a, FT b){ return a + b + std::max(a, b); }; // tighter than 3 * radius
+    // Check if any Voronoi points from ngb need to move to l
+    auto handle_neighbor_voronoi = [&](std::size_t ngb)
+    {
+      auto& ngb_info = landmarks[ngb];
+      auto it = std::remove_if(ngb_info.voronoi.begin(), ngb_info.voronoi.end(), [&](auto wd)
+          {
+            std::size_t w = wd.first;
+            FT d = wd.second;
+            FT newd = dist(l, w);
+            if (newd < d) {
+              if (w != l) // w==l can only happen for ngb==l_parent
+                info.voronoi.emplace_back(w, newd);
+              return true;
+            }
+            return false;
+          });
+      if (it != ngb_info.voronoi.end()) { // modified, always true for ngb==l_parent
+        ngb_info.voronoi.erase(it, ngb_info.voronoi.end());
+        modified_neighbors.push_back(ngb);
+        // We only need to recompute the radius if far was removed, which we can test here with
+        //   if (dist(l, ngb_info.far) < ngb_info.radius)
+        // to avoid a costly test for each w in the loop above, but it does not seem to help.
+        update_radius(ngb);
+        // if (ngb_info.voronoi.empty()) radius_priority.erase(ngb_info.position_in_queue);
+        return true;
+      } else {
+        return false;
+      }
+    };
+    auto handle_neighbor_neighbors = [&](std::size_t ngb)
+    {
+        auto& ngb_info = landmarks[ngb];
+        std::remove_if(ngb_info.neighbors.begin(), ngb_info.neighbors.end(), [&](auto near_){
+            std::size_t near = near_.first;
+            FT d = near_.second;
+            // Conservative 3 * radius: we could use the old radii of ngb and near, but not the new ones.
+            if (d <= 3 * radius) { l_neighbors.insert(near); }
+            // Here it is safe to use the new radii.
+            return d >= max_dist(ngb_info.radius, landmarks[near].radius);
+            });
+    };
+    // First update the Voronoi diagram, so we can compute all the updated
+    // radii before pruning neighbor lists. The main drawback is that we have
+    // to store modified_neighbors, and we don't have access to the old radii
+    // in handle_neighbor_neighbors.
+    handle_neighbor_voronoi(l_parent);
```

Comment:
`handle_neighbor_voronoi` returns true or false, but return type is never used. Should it return `void` ?

Reply:
I guess in an earlier version I was using that boolean. It doesn't hurt, but removed.